lib/URI.pm

```diff
@@ -571,8 +572,12 @@ removing the explicit port specification if it matches the default port,
 uppercasing all escape sequences, and unescaping octets that can be
 better represented as plain characters.
 
-For efficiency reasons, if the $uri is already in normalized form,
-then a reference to it is returned instead of a copy.
+Up to version 1.74, this method would return the original unchanged
+C<$uri> object if it detected nothing to change. To make the return
+value consistent (and since the efficiency gains from this behaviour
+were marginal), this method now unconditionally returns a clone. This
+means idioms like C<$uri-E<gt>clone-E<gt>canonical> are no longer
```

Comment:
This can be shorter (and easier to read in the source) as `C<< $uri->clone->canonical >>`